diesel/src/deserialize.rs

```diff
@@ -545,7 +546,12 @@ where
         use crate::row::Field;
 
         let field = row.get(0).ok_or(crate::result::UnexpectedEndOfRow)?;
-        T::from_nullable_sql(field.value())
+        Ok(T::from_nullable_sql(field.value()).map_err(|e| {
+            Box::new(DeserializeFieldError {
```

Comment:
The CI error messages indicate that I missed something. We use the error type to transport the information that the field is null:

https://github.com/diesel-rs/diesel/blob/ad8ef479e1c6f926b940ad9a43877fc5a993ec5f/diesel/src/deserialize.rs#L443

This means we need to change this mapping here to exclude this specific errors:

This should result in something like:

```rust
.map_err(|e| {
    if !e.is::<crate::result::UnexpectedNullError>() {
        Box::new(DeserializeFieldError {
                field_name: field.field_name().map(|s| s.to_string()),
                error: e,
          })
    } else {
        e
    }
}
```

This should hopefully fix the CI issues,